File: leetCode/Java/q231PowerofTwo.java
/*

https://leetcode.com/problems/power-of-two/

*/
public class q231PowerofTwo {
    
    public static boolean isPowerOfTwo(int n) {
        if (n <= 0)
            return false;
        while (n > 1) {
            if (n % 2 == 1) {
                return false;
            }
            n = n >> 1;
            System.out.println(n);
        }
        return true;
    }
}

File: leetCode/C#/q231PowerofTwo.cs
/*
https://leetcode.com/problems/power-of-two/
*/

public class Solution
{
    public bool IsPowerOfTwo(int n)
    {
        if (n <= 0) return false;
        while (n > 1)
        {
            if (n % 2 == 1)
            {
                return false;
            }
            n = n >> 1;
        }
        return true;

    }
}
